beacon_node/http_api/tests/broadcast_validation_tests.rs

```diff
@@ -0,0 +1,1289 @@
+            BlockStrategy::OnCanonicalHead,
+            AttestationStrategy::AllValidators,
+        )
+        .await;
+    tester.harness.advance_slot();
+
+    let slot_a = Slot::new(num_initial);
+    let slot_b = slot_a + 1;
+
+    let state_a = tester.harness.get_current_state();
+    let (block_a, state_after_a): (SignedBlindedBeaconBlock<E>, _) = tester
+        .harness
+        .make_blinded_block(state_a.clone(), slot_b)
+        .await;
+    let (block_b, state_after_b): (SignedBlindedBeaconBlock<E>, _) =
+        tester.harness.make_blinded_block(state_a, slot_b).await;
+
+    /* check for `make_blinded_block` curios */
+    assert_eq!(block_a.state_root(), state_after_a.tree_hash_root());
+    assert_eq!(block_b.state_root(), state_after_b.tree_hash_root());
+    assert_ne!(block_a.state_root(), block_b.state_root());
+
+    /* submit `block_a` as valid */
+    assert!(tester
+        .client
+        .post_beacon_blinded_blocks_v2(&block_a, validation_level)
+        .await
+        .is_ok());
+
+    /* submit `block_b` which should induce equivocation */
+    let response: Result<(), eth2::Error> = tester
+        .client
+        .post_beacon_blinded_blocks_v2(&block_b, validation_level)
+        .await;
+    assert!(response.is_err());
+
+    let error_response: eth2::Error = response.err().unwrap();
+
+    assert_eq!(error_response.status(), Some(StatusCode::BAD_REQUEST));
+
+    assert!(
+        matches!(error_response, eth2::Error::ServerMessage(err) if err.message ==  "BAD_REQUEST: RepeatProposal { proposer: 44, slot: Slot(32) }")
+    );
+}
+
+/// This test checks that a block that is valid from both a gossip and consensus perspective is accepted when using `broadcast_validation=consensus`.
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+pub async fn blinded_consensus_full_pass() {
+    /* this test targets gossip-level validation */
+    let validation_level: Option<BroadcastValidation> = Some(BroadcastValidation::Consensus);
+
+    // Validator count needs to be at least 32 or proposer boost gets set to 0 when computing
+    // `validator_count // 32`.
+    let validator_count = 64;
+    let num_initial: u64 = 31;
+    let tester = InteractiveTester::<E>::new(None, validator_count).await;
+
+    // Create some chain depth.
+    tester.harness.advance_slot();
+    tester
+        .harness
+        .extend_chain(
+            num_initial as usize,
+            BlockStrategy::OnCanonicalHead,
+            AttestationStrategy::AllValidators,
+        )
+        .await;
+    tester.harness.advance_slot();
+
+    let slot_a = Slot::new(num_initial);
+    let slot_b = slot_a + 1;
+
+    let state_a = tester.harness.get_current_state();
+    let (block, _): (SignedBlindedBeaconBlock<E>, _) =
+        tester.harness.make_blinded_block(state_a, slot_b).await;
+
+    let response: Result<(), eth2::Error> = tester
+        .client
+        .post_beacon_blinded_blocks_v2(&block, validation_level)
```

Comment:
I'm kinda surprised that the BN is able to unblind this block, but if it works, it works! There's a cache in the execution layer that keeps track of block bodies for blinded blocks, and I wasn't sure if `make_blinded_block` would prime that cache

Reply:
Marking this as resolved, I think what's happening is that the payload is empty in these tests, so the payload unblinding is trivial. We can address this in future if we like, but for now the blinded case is not so important anyway.